### PR TITLE
Allow certain blocks to be placed on top of upside-down stairs/slabs

### PR DESCRIPTION
**Pull request by havel06**
Pressure plates, repeaters, comparators and redstone wire can now be placed on top of upside-down slabs and stairs.

### CODE REVIEW

File: src/Blocks/BlockComparator.h
Comment (bearbin) [inline src/Blocks/BlockComparator.h:175]: I think this function should be part of the cBlockStairsHandler, like the IsSlabType above.

**Review by bearbin — COMMENTED** [review]
Looks good, just a comment about the location of IsBlockStair.

Also, could you refactor the torch handler to use your new function as well, it uses custom logic at the moment: https://github.com/cuberite/cuberite/blob/a62b2b1be2103d7de2fd66c7304b7473e369be3c/src/Blocks/BlockTorch.h#L23

I would just remove the switch cases for stairs and add a check to your function (and to slabs) in the default case.

File: src/Blocks/BlockComparator.h
Comment (havel06) [reply src/Blocks/BlockComparator.h:175]: True, It would be more consistent with the slab function.

**Comment by havel06** [comment]
I have made changes to buttons, rails and levers, so they can now be placed on top of upside down stairs and slabs as well. Also, torches, levers and buttons can now be placed on the "tall" side of a stair block.

**Review by bearbin — APPROVED** [review]
Looking good, I'll wait for another review but this is ready to merge I think.

**Pull request merged**
